src/wallet.js

```diff
@@ -353,8 +353,14 @@ MyWallet.initializeWallet = function (pw, decryptSuccess, buildHdSuccess) {
     };
     return MyWallet.wallet.loadExternal.bind(MyWallet.wallet)().catch(loadExternalFailed);
   };
-
-  return Promise.resolve().then(doInitialize).then(tryLoadExternal);
+  var incStats = function () {
+    var gh = MyWallet.wallet.getHistory.bind(MyWallet.wallet);
```

Comment:
We can at least make this `getHistory` call more efficient:
1 - only ask the balance, not the full transaction history
2 - only check legacy address
3 - don't check watch-only addresses